apps/examples/testcase/ta_tc/task_manager/itc/itc_task_manager_main.c

```diff
@@ -0,0 +1,300 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, 
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+#include <tinyara/config.h>
+#include <semaphore.h>
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <stdbool.h>
+#include <sys/types.h>
+#include <sys/wait.h>
+#include <task_manager/task_manager.h>
+#ifdef CONFIG_TASK_MANAGER_USER_SPECIFIC_BROADCAST
+#include <task_manager/task_manager_broadcast_list.h>
+#endif
+#include "tc_common.h"
+
+#define TM_SAMPLE_NAME      "tm_sample"
+#define TM_BROADCAST1_NAME  "tm_broadcast1"
+#define TM_BROADCAST2_NAME  "tm_broadcast2"
+#define TM_BROADCAST3_NAME  "tm_broadcast3"
+#define TM_NOPERM_NAME      "tm_noperm"
+#define TM_SAMPLE_MSG       "test_msg"
+#define TM_SYNC_SEND_MSG    "send_sync"
+#define TM_SYNC_RECV_MSG    "recv_sync"
+#define TM_INVALID_HANDLE               -2
+#define TM_INVALID_TIMEOUT              -2
+#define TM_INVALID_PERMISSION           -2
+#define TM_INVALID_MSG_MASK             -2
+#define TM_INVALID_BROAD_MSG            -2
+#define TM_INVALID_PID                  -2
+#define TM_BROAD_TASK_NUM               2
+#define TM_BROAD_UNDEF_NUM              3
+#define TM_BROAD_WIFI_ON_DATA           "BROAD_WIFI_ON"
+#define TM_BROAD_WIFI_OFF_DATA          "BROAD_WIFI_OFF"
+#define TM_BROAD_UNDEFINED_MSG_DATA     "BROAD_UNDEFINED"
+#define TM_BROAD_UNDEFINED_MSG_NOT_USED (TM_BROADCAST_MSG_MAX + 2)
+#define EXIT_CB_CHECK_STR               "exit_cb_chk"
+
+static int tm_sample_handle;
+static int tm_sample_handle2;
+static int tm_noperm_handle;
+static tm_appinfo_list_t *sample_list_info;
+static int pid_tm_itc;
+static int handle_tm_itc;
+static bool cb_flag;
+
+static int not_builtin_task_1(int argc, char *argv[])
+{
+
+	while (1) {
+		usleep(1);
+	}	// This will be dead at itc_task_manager_stop_p
+	return OK;
+}
+
+static void test_stop_cb(void *data)
+{
+	cb_flag = true;
+}
+
+int tm_itc_sample_main(int argc, char *argv[])
+{
+	int ret;
+
+	tm_noperm_handle = task_manager_register_builtin(TM_NOPERM_NAME, TM_APP_PERMISSION_DEDICATE, 100);
+
+	ret = task_manager_set_stop_cb(test_stop_cb, NULL);
+	if (ret != OK) {
+		cb_flag = false;
+	}
+
+	while (1) {
+		usleep(1);
+	}
+	return ret;
+}
+
+void itc_task_manager_check_dedicate_permission_p(void)
+{
+	tm_sample_handle = task_manager_register_builtin(TM_SAMPLE_NAME, TM_APP_PERMISSION_DEDICATE, TM_RESPONSE_WAIT_INF);
+	TC_ASSERT_GEQ("task_manager_register", tm_sample_handle, 0);
+
+	tm_sample_handle2 = task_manager_register_builtin("itc_task_manager_user1", TM_APP_PERMISSION_DEDICATE, TM_RESPONSE_WAIT_INF);
+	TC_ASSERT_GEQ("task_manager_register", tm_sample_handle2, 0);
+
+	int ret;
+	ret = task_manager_start(tm_sample_handle2, TM_RESPONSE_WAIT_INF);
+	TC_ASSERT_EQ("task_manager_start", ret, OK);
```

Comment:
Is it okay not to unregister the registered task when task_manager_start fail case?